Use rucio policy package

dynamic-entropy opened the pull request:
Builds on top of: Setting python path for Policy packages: https://github.com/dmwm/CMSRucio/pull/357

Goes together with: Flux setting for configuring the policy package environment: https://github.com/dmwm/rucio-flux/pull/179

Required for: Using the auto-approval plugin: https://github.com/rucio/rucio/issues/6203

Fixes: Move CMS to policy package: Fix https://github.com/dmwm/CMSRucio/issues/250

----------

- The use of relative path in `COPY` locks us to use the `.` docker build context. Should not be a big trouble. Any other suggestions? 
- Refactored the organisation of policy package:
```
src/policy/CMSRucioPolicy
|-- __init__.py
|-- permission.py
|-- schema.py
|-- algorithms
    |-- algorithm_class1.py
    |-- algorithm_class2.py
```

- Relocated and defined `cmstfc` in the `lfn2pfn algorithm class
- We will add the `auto_approval` class here soon

ericvaandering commented on docker/rucio-server/Dockerfile:
```diff
@@ -60,12 +48,12 @@ RUN mkdir -p /patch
 # My patch for FNAL URL problem
 ADD https://patch-diff.githubusercontent.com/raw/rucio/rucio/pull/6097.patch /patch/6097.patch
-# Rahul patch for account name problem
+# Rahul patch for account name problem #FIXME: Also this one can be removed
```
If we don't need these things, please remove them. Also line 57.

ericvaandering commented on src/policy/CMSRucioPolicy/__init__.py:
```diff
@@ -0,0 +1,18 @@
+
+from CMSRucioPolicy.algorithms import lfn2pfn
+
+SUPPORTED_VERSION = ["1.30"]
```
I'm going to put this into 1.31, not 1.30.

dynamic-entropy replied:
Added 1.31 too.

ericvaandering commented on src/policy/CMSRucioPolicy/algorithms/lfn2pfn.py:
```diff
@@ -0,0 +1,199 @@
+#! /usr/bin/env python3
```
This file is just a copy of something we already have, right? Can you use git mv so that we keep the history of that file?

dynamic-entropy replied:
Used `git mv` on relevant files. 
Later modified them as needed. 

Thankyou for the suggestion.

ericvaandering commented:
OK. Changes in the code asked for. I have two others. Please update the release in rucio-server/build-images.sh while we are still transitioning. And you will need changes to the daemons, probes, and webui docker files too.

Also please squash your commits or rewrite them so they are few and logical.

dynamic-entropy commented:
Hi @ericvaandering 

> Done. Please update the release in `rucio-server/build-images.sh` while we are still transitioning. 

Did you mean (https://github.com/dmwm/CMSRucio/pull/517) by transitioning?
Also, updated the script.

> And you will need changes to the daemons, probes, and webui docker files too.

Done. Thank you.

Squashed the commits into three logical categories of updates.
